### PR TITLE
2 x 42151 jaggaer email engagement

### PR DESCRIPTION
**Pull request by azwansyafiq-2X**
Standardization for Jaggaer email engagement script. 
Remove subquery, adjust styling.

### CODE REVIEW

**Comment by nazzatul21** [comment]
noting to comment all good :)

**Review by raaayden — APPROVED** [review]
everything looks fine so far. Perhaps it will be better for the prospects to move into a separate script which is tam_database_log

File: Clients/Jaggaer/Analytics Script/Normalization_Script/Email_Engagement/jaggaer_db_email_engagement_log.sql [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+    download._prospectID,
+    download._email,
+    click._campaignID,
+    click._utmcampaign,
+    download._timestamp,
+    download._description,
+    download._engagement,
+    click._email_id,
+    click.email_template_id,
+  FROM (
```
Comment (izztsfn) [inline Clients/Jaggaer/Analytics Script/Normalization_Script/Email_Engagement/jaggaer_db_email_engagement_log.sql:388]: Create a new CTE to remove subquery

File: Clients/Jaggaer/Analytics Script/Normalization_Script/Email_Engagement/jaggaer_db_email_engagement_log.sql [hunk 2]
```diff
@@ -0,0 +1,613 @@
...
+    SELECT * 
+    FROM clicks_downloads_timeline
+    WHERE _engagement = 'Downloaded'
+  ) download
+  JOIN (
+    SELECT *
```
Comment (izztsfn) [inline Clients/Jaggaer/Analytics Script/Normalization_Script/Email_Engagement/jaggaer_db_email_engagement_log.sql:394]: Create a new CTE here to remove subquery

File: Clients/Jaggaer/Analytics Script/Normalization_Script/Email_Engagement/jaggaer_db_email_engagement_log.sql [hunk 3]
```diff
@@ -0,0 +1,613 @@
...
+) scenario
+WHERE origin._prospectID = scenario._prospectID
+  AND origin._email = scenario._email
+  AND origin._campaignID = scenario._campaignID
+  AND origin._list_email_id = scenario._list_email_id
+  AND TIMESTAMP_DIFF(click_timestamp, open_timestamp, SECOND) < 3
+;
+
+
+CREATE OR REPLACE TABLE `x-marketing.jaggaer.db_lead_campaign_pardot` AS
```
Comment (izztsfn) [inline Clients/Jaggaer/Analytics Script/Normalization_Script/Email_Engagement/jaggaer_db_email_engagement_log.sql:510]: Use DML statement instead

**Review by izztsfn — CHANGES_REQUESTED** [review]
Overall formatting looks great! Just some subqueries to remove and also replace Create statement with Truncate and Insert instead.

Also i hope you have checked the rows of this script to match the existing tables. This is to ensure that the formatting does not affect the existing solution. So a quick check wouldve been sufficient. Can refer [this slide, number 6](https://2xmarketing-my.sharepoint.com/:p:/g/personal/izzat_sofian_2x_marketing/Ed-Y15u-YSZOtl_V4L6eZ6gBFByJHCck1C-5E5DvDiYdAg?e=odot3p&nav=eyJzSWQiOjMwMSwiY0lkIjoxMzk1MzcxNTR9) i made last time

**Review by izztsfn — APPROVED** [review]
Noice!

**Comment by izztsfn** [comment]
@azwansyafiq-2X dont forget to update the approved script into the scheduler ya tq!!

**Pull request merged**